lib/cmd/common/initlocal.js

```diff
@@ -256,59 +266,39 @@ function doInitLocal(projectId, appId, cb) {
   log.silly('Getting app container from url: ' + fhreq.getFeedHenryUrl() + ", domain: " + fhc.curTarget, 'get container');
 
   // check valid appid
-  common.readApp(projectId, appId, function (err) {
-    if (err) {
-      return cb(err);
-    }
-    installFhDbLocally(function (err) {
-      if (err) {
-        return cb(err);
-      }
-      createLocalDirectory(localDirectory, function (err) {
+  return async.waterfall([
+    common.readApp.bind(this, projectId, appId),
+    installFhDbLocally,
+    createLocalDirectory.bind(this, localDirectory),
+    checkApplicationJS.bind(this, applicationJS),
+    common.getAppContainer.bind(this, appId),
```

Comment:
@paolobueno then better to use promises instead if they were using variables available from the closure as params to direct calls. But we can leave it for now/Add please some TODO that we need to change this to use promise instead